src/sensai/featuregen.py

```diff
@@ -24,28 +24,45 @@ class FeatureGenerator(ABC):
     from an input DataFrame
     """
     def __init__(self, categoricalFeatureNames: Sequence[str] = (),
-            normalisationRules: Sequence[data_transformation.DFTNormalisation.Rule] = (), addCategoricalDefaultRules=True):
+                 normalisationRules: Sequence[data_transformation.DFTNormalisation.Rule] = (),
+                 generatedColumnsRuleTemplate: data_transformation.DFTNormalisation.RuleTemplate = None, addCategoricalDefaultRules=True):
         """
         :param categoricalFeatureNames: if provided, will ensure that the respective columns in the generated data frames will
             have dtype 'category'.
             Furthermore, presence of meta-information can later be leveraged for further transformations, e.g. one-hot encoding.
         :param normalisationRules: Rules to be used by DFTNormalisation (e.g. for constructing an input transformer for a model).
             These rules are only relevant if a DFTNormalisation object consuming them is instantiated and used
             within a data processing pipeline. They do not affect feature generation.
+        :param generatedColumnsRuleTemplate: If given, a normalisation rule based on this template will be created for all generated columns.
+            As with other rules, this is only relevant if DFTNormalization is used after feature generation.
+            The rule template should only be provided if no categoricalFeatureNames and normalisationRules have been passed
         :param addCategoricalDefaultRules:
             If True, normalisation rules for categorical features (which are unsupported by normalisation) and their corresponding one-hot
             encoded features (with "_<index>" appended) will be added.
         """
+        self.generatedColumnNames = None
+        self._generatedColumnsRule = None
+        self.generatedColumnsRuleTemplate = generatedColumnsRuleTemplate
         self._categoricalFeatureNames = categoricalFeatureNames
         normalisationRules = list(normalisationRules)
         if addCategoricalDefaultRules and len(categoricalFeatureNames) > 0:
             normalisationRules.append(data_transformation.DFTNormalisation.Rule(r"(%s)" % "|".join(categoricalFeatureNames), unsupported=True))
             normalisationRules.append(data_transformation.DFTNormalisation.Rule(r"(%s)_\d+" % "|".join(categoricalFeatureNames), skip=True))
         self._normalisationRules = normalisationRules
 
+        self._inputValidation()
+
+    def _inputValidation(self):
+        if self.generatedColumnsRuleTemplate is not None:
+            if self.getNormalisationRules():
+                raise ValueError(f"normalisationRules should be empty when a generatedColumnsRuleTemplate is provided, instead got {self.getNormalisationRules()}")
+
     def getNormalisationRules(self) -> Sequence[data_transformation.DFTNormalisation.Rule]:
         return self._normalisationRules
 
+    def getGeneratedColumnsRule(self):
```

Comment:
Why do we need this?

Reply:
We don't really need it. I thought it might be useful as info for the user (to see what rule a generator has applied once it ran) but probably it is more confusing than helping. I'll remove it and add a getter for the ruleTemplate instead